content/blog/gain-platform-superpowers-with-the-kebap-stack/index.md

```diff
@@ -0,0 +1,704 @@
+  project: ${{ (values.owner | parseEntityRef).name }}
+  source:
+    path: "gitops/teams/clusters/${{values.name}}-${{values.stage}}-cluster"
+    directory:
+      recurse: false
+      exclude: '{catalog-info.yaml}'
+    targetRevision: main
+    repoURL: https://github.com/my-backstage-demo/backstage-infrastructure-provisioning-templates-workshop.git
+  syncPolicy:
+    automated:
+      prune: true
+      selfHeal: true
+    syncOptions:
+    - ServerSideApply=true
+    - CreateNamespace=true
+```
+
+### Pulumi
+
+<img src="p.png" >
+
+Pulumi manages the lifecycle of all the cloud infrastructure, developers self-service. It allows you to use familiar
+programming languages to define your infrastructure as code, providing more flexibility and power than traditional YAML
+or domain-specific languages. The [Pulumi Kubernetes Operator](https://github.com/pulumi/pulumi-kubernetes-operator)
+enables you to manage your infrastructure directly from within Kubernetes, offering seamless integration with your
+existing Kubernetes workflows.
+
+As we use GitOps to manage the infrastructure, we define the infrastructure in the Git repository let
+then [Argo CD](#argo-cd) deploy it to the cluster. As soon the definition of the infrastructure is deployed to the
+cluster, the Pulumi Kubernetes Operator will take care and create the defined resources in the selected cloud provider.
+
+Here is an example of how we can use the Pulumi Kubernetes Operator to create a new Kubernetes cluster
+in `DigitalOcean`:
+
+```yaml
+apiVersion: pulumi.com/v1
+kind: Program
+metadata:
+  name: ${{values.name}}
+program:
+  variables:
+    clusterName: "do-cluster"
+    clusterRegion: "nyc1"
+    nodePoolName: "default"
+    nodeCount: 1
+    version: 1.30.1-do.0
+
+  resources:
+    do_cluster:
+      type: digitalocean:KubernetesCluster
+      properties:
+        name: "${clusterName}"
+        region: "${clusterRegion}"
+        version: "${version}"
+        destroyAllAssociatedResources: true
+        nodePool:
+          name: "${nodePoolName}"
+          size: "s-2vcpu-2gb"
+          nodeCount: "${nodeCount}"
+    # cut for brevity
+---
+apiVersion: pulumi.com/v1
+kind: Stack
+metadata:
+  name: ${{values.name}}
+spec:
+  stack: ${{values.stack}}
+  envRefs:
+    DIGITALOCEAN_TOKEN:
+      type: Env
+      env:
+        name: DIGITALOCEAN_TOKEN
+  programRef:
+    name: ${{values.name}}
+  destroyOnFinalize: false
+```
+
+### External Secrets Operator (ESO)
+
+<img src="e.png" >
+
+ESO addresses the challenge of securely managing secrets in Kubernetes. It integrates with various secret stores (such
+as AWS Secrets Manager, HashiCorp Vault, Google Secrets Manager, Azure Key Vault) in a consistent way. This allows you
+to configure access to the secret store once for all your required secrets, enhancing security and simplifying
+management.
+
+We even take it a step further by using ESO to manage the secrets from Pulumi ESC (External Secret Configurations) with
```

Comment:
```suggestion
We even take it a step further by using ESO to manage the secrets from [Pulumi ESC (Environments, Secrets, and  Configurations)](https://www.pulumi.com/product/esc/) with
```

There is a typo, it is Environments, Secrets, and Configuration. Also please always crosslink our products